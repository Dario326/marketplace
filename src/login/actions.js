import * as api from "../api/loginApi";
import { LocalStorageHelper } from "../utilities";

const LOGIN = "login/";
export const IS_LOADING = `${LOGIN}IS_LOADING`;
export const LOGIN_SUCCESS = `${LOGIN}LOGIN_SUCCESS`;
export const LOGIN_ERROR = `${LOGIN}LOGIN_ERROR`;
export const LOGOUT = `${LOGIN}LOGOUT`;

export function loginSuccess() {
    return { type: LOGIN_SUCCESS, login };
}

export function loginError() {
    return { type: LOGIN_ERROR, login };
}

export function toggleLoading(status = false) {
    return { type: IS_LOADING, status };
}

export function logout() {
    window.history.pushState(null, null, window.location.href);
    window.onpopstate = function () {
        window.history.go(1);
    };
    return { type: LOGOUT, logout };
}



export function login(username, password) {
    return function (dispatch) {
        dispatch(toggleLoading(true));
        return api.login(username, password).then(response => {
            if (response.result === "SUCCESS") {
                dispatch(loginSuccess(response.result));
                dispatch(toggleLoading(false));
            } else {
                dispatch(loginError());
                dispatch(toggleLoading(false));
                throw response.message;
            }
        });
    };
}



