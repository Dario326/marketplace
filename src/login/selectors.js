export const getUserData = (state) => {
    return state.LOGIN.userData;
};
export const getSettingsData = (state) => {
    return state.LOGIN.settingsData;
};

export const getAuthenticated = (state) => {
    return state.LOGIN.isAuthentcated;
};