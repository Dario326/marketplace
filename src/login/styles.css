.login-container{
    display: flex;
    flex-direction: column;
    justify-content: center;
    height: 100vh;
    align-items: center;
}
.login-screen{
    max-width: 400px;
}
.login-logo{
    border-bottom-style: solid;
    /* border-color: #A28D5B; */
    border-width: 3px;    
    -webkit-border-image: 
      -webkit-gradient(linear, 0 0, 0 100%, from(#A28D5B), to(rgba(0, 0, 0, 0))) 100% 1;
    -webkit-border-image: 
      -webkit-linear-gradient(#A28D5B, rgba(0, 0, 0, 0)) 100% 1;
    -moz-border-image:
      -moz-linear-gradient(#A28D5B, rgba(0, 0, 0, 0)) 100% 1;
    -o-border-image:
      -o-linear-gradient(#A28D5B, rgba(0, 0, 0, 0)) 100% 1;
    border-image:
      linear-gradient(to right, #A28D5B, rgba(0, 0, 0, 0)) 100% 1;
}
.forgotPassword{
    color: #A28D5B;
}
.relative{
  position: relative;
}
.loginBtn{
  opacity: 0.4;
  position: relative;
}
.loginBtn:after{
  content: "";
  position: absolute;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
}
.loginBtn.active{
  opacity: 1;
}
.loginBtn.active:after{
  content: "";
  display: none
}
.loginLoader{
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
}