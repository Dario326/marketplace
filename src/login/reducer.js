
import * as actions from "./actions";
export const LOGIN_REDUCER_KEY = "LOGIN";

export function reducer(state = { }, action) {
    switch (action.type) {
        case actions.LOGIN_SUCCESS: {
            return {
                ...state,
                isAuthentcated: true,
                isLoginError: false
            };
        }
        case actions.LOGIN_ERROR:
            return {
                ...state,
                isLoginError: true,
                isAuthentcated: true
            };
        case actions.IS_LOADING: {
            return {
                ...state,
                isLoading: action.status
            };
        }
        case actions.LOGOUT: {
            return {
                ...state,
                isLoginError: false,
                isAuthentcated: false,
                isLoading: false
            };
        }
        default:
            return state;
    }
}
