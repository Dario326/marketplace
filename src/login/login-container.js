import { Login as LoginPage } from "./login";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import * as actions from "../login/actions";
import { getUserData, getSettingsData } from "./selectors";

const mapStateToProps = (state) => {
    return {
        isLoading: state.LOGIN.isLoading,
        isAuthentcated: state.LOGIN.isAuthentcated,
        isLoginError: state.LOGIN.isLoginError,
        userData: getUserData(state),
        settingsData: getSettingsData(state)
    };
};

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators(actions, dispatch)
    };
}
export const Login = connect(mapStateToProps, mapDispatchToProps)(LoginPage);