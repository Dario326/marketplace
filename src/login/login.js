import React from "react";
import PropTypes from "prop-types";
import { NavLink } from "react-router-dom";
import "./styles.css";

export class Login extends React.Component {
    static propTypes = {
        actions: PropTypes.object,
        isLoading: PropTypes.bool,
        isAuthentcated: PropTypes.bool,
        isLoginError: PropTypes.bool,
        history: PropTypes.object,
        userData: PropTypes.object,
        settingsData: PropTypes.object
    }
    constructor(props,context) {
        super(props,context);
        this.state = {
            userName: "",
            passWord: "",
            isShowlogin: false,
        };
        this.userNameCheck = this.userNameCheck.bind(this);
        this.passwordCheck = this.passwordCheck.bind(this);
    }
    componentWillReceiveProps() {
        if (this.props.isAuthentcated && !this.props.isLoginError) this.props.history.push("/");
    }
    userNameCheck = (event) => {
        this.setState({ userName: event.target.value },this.showLogin);
    }
    passwordCheck = (event) => {
        this.setState({ passWord: event.target.value },this.showLogin);
    }
    showLogin = () => {
        const { userName, passWord } = this.state;
        if (userName.length > 0 && passWord.length > 0) this.setState({ isShowlogin: true });
        else this.setState({ isShowlogin: false });
    }
    render() {
        let { userName, passWord } = this.state;
        let { isLoading, isLoginError } = this.props;
        const loginBtn = this.state.isShowlogin ? "form-group loginBtn active" : "form-group loginBtn";
        return (
            <div className="text-center login-container animated fadeInDown border-size-md ">
                <div className="login-screen gray-bg p-m b-r-xl">
                    <div className="text-center p-xs border-size-xs login-logo">
                        {/*<img className="img-responsive" style={{ marginLeft: "50px" }} src={require("../img/GTpay.png")} alt=""/>*/}
                    </div>
                    <div className="p-m">
                        <h3>Welcome to the Marketplace</h3>
                        <p>Please enter your username and password to log in.</p>
                        <div className="m-t relative">
                            { isLoading && <div className="alert alert-info loginLoader"><i class="fa fa-spinner fa-pulse fa-lg m-r-xs"/> Please wait...</div> }
                            {/* { isAuthentcated && <div className="alert alert-success"><i class="fa fa-check m-r-xs fa-lg"/> Successfully logging in</div>} */}
                            { isLoginError && <div className="alert alert-danger"><i class="fa fa-close m-r-xs fa-lg"/> Please enter valid Username / Password</div> }
                            <div className="form-group">
                                <input type="text" onChange={this.userNameCheck} className="form-control" placeholder="Username"/>
                            </div>
                            <div className="form-group">
                                <input type="password" onChange={this.passwordCheck} className="form-control" placeholder="Password"/>
                            </div>
                            <div className={loginBtn}>
                                <button type="submit" className="dim btn btn-primary block full-width m-b" onClick={() => this.props.actions.login(userName,passWord)}>Login</button>
                            </div>
                            <div className="form-group text-right">
                                <NavLink className=" forgotPassword" to="forgot_password"><small>Forgot password?</small></NavLink>
                            </div>
                            <div className="form-group text-right">
                                <NavLink className=" forgotPassword" to="forgot_password"><small>Sign Up</small></NavLink>
                            </div>
                        </div>

                        <p className="m-t"> <small>Marketplace &copy; 2018</small> </p>

                    </div>
                </div>
            </div>
        );
    }
}

Login.contextTypes = {
    router: PropTypes.object
};