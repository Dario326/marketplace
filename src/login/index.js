import { Login } from "./login-container";

export {
    Login
};