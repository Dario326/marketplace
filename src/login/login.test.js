import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router";
import { Login } from "./login";

describe("<Login />", () => {
    describe("Rendering...", () => {
        it("Should match <Login /> snapshot", () => {
            const component = renderer.create(
                <MemoryRouter>
                    <Login />
                </MemoryRouter>,
            );
            let tree = component.toJSON();
            expect(tree).toMatchSnapshot();
        });

        it("Renders No username error message", () => {
            // Write a test to check if error message
        });

        it("Renders No password error message", () => {
            // Write a test to check if error message
        });

        it("Renders No username and password error message", () => {
            // Write a test to check if error message
        });
    });

    describe("User login", () => {
        it("Stores token to the localstorage for caching post successful login", () => {
            // Write a test to check if localstorage is initialized
        });
    });
});
