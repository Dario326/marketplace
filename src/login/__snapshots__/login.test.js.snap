// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> Rendering... Should match <Login /> snapshot 1`] = `
<div>
  <div
    className="text-center"
  >
    <h1
      className="logo-name"
    >
      marketplace
    </h1>
  </div>
  <div
    className="middle-box text-center loginscreen animated fadeInDown"
  >
    <div>
      <h3>
        Welcome to the Marketplace
      </h3>
      <p>
        Please enter your username and password to log in.
      </p>
      <form
        className="m-t"
        role="form"
      >
        <div
          className="form-group"
        >
          <input
            className="form-control"
            placeholder="Username"
            required=""
            type="email"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            placeholder="Password"
            required=""
            type="password"
          />
        </div>
        <button
          className="btn btn-primary block full-width m-b"
          type="submit"
        >
          Login
        </button>
        <a
          aria-current={false}
          className={undefined}
          href="forgot_password"
          onClick={[Function]}
          style={undefined}
        >
          <small>
            Forgot password?
          </small>
        </a>
        <p
          className="text-muted text-center"
        >
          <small>
            Do not have an account?
          </small>
        </p>
        <p
          className="text-muted text-center"
        >
          Contact Sales Or Your Account Manager
        </p>
      </form>
      <p
        className="m-t"
      >
         
        <small>
          Konnektive Corp. © 2018
        </small>
         
      </p>
    </div>
  </div>
</div>
`;

exports[`Should match <Login /> snapshot 1`] = `
<div>
  <div
    className="text-center"
  >
    <h1
      className="logo-name"
    >
      Marketplace
    </h1>
  </div>
  <div
    className="middle-box text-center loginscreen animated fadeInDown"
  >
    <div>
      <h3>
        Welcome to Marketplace
      </h3>
      <p>
        Please enter your username and password to log in.
      </p>
      <form
        className="m-t"
        role="form"
      >
        <div
          className="form-group"
        >
          <input
            className="form-control"
            placeholder="Username"
            required=""
            type="email"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            placeholder="Password"
            required=""
            type="password"
          />
        </div>
        <button
          className="btn btn-primary block full-width m-b"
          type="submit"
        >
          Login
        </button>
        <a
          aria-current={false}
          className={undefined}
          href="forgot_password"
          onClick={[Function]}
          style={undefined}
        >
          <small>
            Forgot password?
          </small>
        </a>
        <p
          className="text-muted text-center"
        >
          <small>
            Do not have an account?
          </small>
        </p>
        <p
          className="text-muted text-center"
        >
          Contact Sales Or Your Account Manager
        </p>
      </form>
      <p
        className="m-t"
      >
         
        <small>
          Konnektive Corp. © 2018
        </small>
         
      </p>
    </div>
  </div>
</div>
`;
